pytorch_lightning/trainer/trainer.py

```diff
@@ -516,8 +516,14 @@ def dispatch(self):
         else:
             self.accelerator.start_training(self)
 
+    def on_run_stage_setup(self):
+        self.profiler.setup(local_rank=self.local_rank if self.world_size > 1 else None, log_dir=self.log_dir)
+
```

Comment:
if it's not meant to be called outside the trainer
```suggestion
    def _on_run_stage_setup(self):
        self.profiler.setup(local_rank=self.local_rank if self.world_size > 1 else None, log_dir=self.log_dir)

```

Reply:
Good catch ! Definitely.